californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -197,33 +202,33 @@ public int assignMessageId(final Message message) {
 	}
 
 	private int registerWithMessageId(final Exchange exchange, final Message message) {
-
 		enableStatus = true;
+		exchange.assertIncomplete(message);
```

Comment:
If a Exchange is complete, it would not clean-up the store again. Therefore I assert the incompleteness (and assert the ownership with that). The point here is, that if this is violated somewhere, it's easier to assert here and see, where the violation is from, then do a analysis afterwards.

Reply:
Checking the owner should detect "concurrent" execution.

Reply:
it makes sense